### PR TITLE
[doc] Improve Pulsar Security Kerberos

### PR DESCRIPTION
**Pull request by Monica-zy**
Motivation
Improve the language and the overall descriptive style of the Pulsar Security document ( Security-kerberos section): http://pulsar.apache.org/docs/en/next/security-kerberos/

Modifications
Adjust the tone, personal pronouns, voice also some typo errors of some sentences in the document.

### CODE REVIEW

**Comment by Monica-zy** [comment]
@Jennifer88huang  Please review, thank you~

**Comment by Monica-zy** [comment]
run java8 tests

File: site2/docs/security-kerberos.md [hunk 1]
```diff
@@ -4,26 +4,26 @@ title: Authentication using Kerberos
 sidebar_label: Authentication using Kerberos
 ---
 
-[Kerberos](https://web.mit.edu/kerberos/) is a network authentication protocol. It is designed to provide strong authentication for client/server applications by using secret-key cryptography. 
+[Kerberos](https://web.mit.edu/kerberos/) is a network authentication protocol. By using secret-key cryptography, [Kerberos](https://web.mit.edu/kerberos/) is designed to provide strong authentication for client applications or server applications.
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]: or --> and

File: site2/docs/security-kerberos.md [hunk 2]
```diff
@@ -4,26 +4,26 @@ title: Authentication using Kerberos
...
+[Kerberos](https://web.mit.edu/kerberos/) is a network authentication protocol. By using secret-key cryptography, [Kerberos](https://web.mit.edu/kerberos/) is designed to provide strong authentication for client applications or server applications. 
 
-In Pulsar, we use Kerberos with [SASL](https://en.wikipedia.org/wiki/Simple_Authentication_and_Security_Layer) as a choice for authentication. And Pulsar uses the [Java Authentication and Authorization Service (JAAS)](https://en.wikipedia.org/wiki/Java_Authentication_and_Authorization_Service) for SASL configuration. You must provide JAAS configurations for Kerberos authentication. 
+In Pulsar, we use Kerberos with [SASL](https://en.wikipedia.org/wiki/Simple_Authentication_and_Security_Layer) as a choice for authentication. And Pulsar uses the [Java Authentication and Authorization Service (JAAS)](https://en.wikipedia.org/wiki/Java_Authentication_and_Authorization_Service) for SASL configuration. You need to provide JAAS configurations for Kerberos authentication.
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]: Try to avoid to use "we", "I", "us" in technical writing.

File: site2/docs/security-kerberos.md [hunk 3]
```diff
@@ -4,26 +4,26 @@ title: Authentication using Kerberos
...
+To begin, you need to set up (or already have) a [Key Distribution Center(KDC)](https://en.wikipedia.org/wiki/Key_distribution_center). Also you need to configure and run the [Key Distribution Center(KDC)](https://en.wikipedia.org/wiki/Key_distribution_center)in advance. 
 
-If your organization is already using a Kerberos server (for example, by using `Active Directory`), there is no need to install a new server for Pulsar. Otherwise you will need to install one. Your Linux vendor likely has packages for `Kerberos` and a short guide on how to install and configure it: ([Ubuntu](https://help.ubuntu.com/community/Kerberos), 
+If your organization is already using a Kerberos server (for example, by using `Active Directory`), you do not have to install a new server for Pulsar. If your organization is not using a Kerberos server, you need to install one. Your Linux vendor might have packages for `Kerberos`. Refer to a short guide on how to install and configure it from here: ([Ubuntu](https://help.ubuntu.com/community/Kerberos),
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]: On how to install and configure Kerberos, refer to ....

File: site2/docs/security-kerberos.md [hunk 4]
```diff
@@ -391,5 +387,5 @@ And add a section of `BookKeeper` configurations in the `pulsar_jaas.conf` that
 };
 ```
 
-In this setting, Pulsar Broker's principal and keyTab file indicates Broker's role when authenticating with Bookie.
+In this setting, the principal of Pulsar Broker and keyTab file indicates the role of Broker when you are authenticating with Bookie.
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]:
```suggestion
In this setting, the principal of Pulsar Broker and keyTab file indicates the role of Broker when you authenticate with Bookie.
```

File: site2/docs/security-kerberos.md [hunk 5]
```diff
@@ -366,19 +362,19 @@ And add a section of `Client` configurations in the file `pulsar_jaas.conf`, whi
...
+Pulsar Broker acts as a Kerberos client when you are authenticating with Bookie. According to [BookKeeper document](http://bookkeeper.apache.org/docs/latest/security/sasl/), you need to add `bookkeeperClientAuthenticationPlugin` parameter in `broker.conf`:
 
 ```
 bookkeeperClientAuthenticationPlugin=org.apache.bookkeeper.sasl.SASLClientProviderFactory
 ```
 
 In this setting, `SASLClientProviderFactory` creates a BookKeeper SASL client in a Broker, and the Broker uses the created SASL client to authenticate with a Bookie node.
 
-And add a section of `BookKeeper` configurations in the `pulsar_jaas.conf` that used by Pulsar Broker:
+And enter the following commands to add a section of `BookKeeper` configurations in the `pulsar_jaas.conf` that Pulsar Broker uses:
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]: Delete "And" at the beginning.

File: site2/docs/security-kerberos.md [hunk 6]
```diff
@@ -366,19 +362,19 @@ And add a section of `Client` configurations in the file `pulsar_jaas.conf`, whi
 };
 ```
 
-In this setting, Pulsar Broker's principal and keyTab file indicates Broker's role when authenticating with ZooKeeper.
+In this setting, the principal of Pulsar Broker and keyTab file indicates the role of Broker when you are authenticating with ZooKeeper.
 
 ## Regarding authentication between BookKeeper and Broker
 
-Pulsar Broker acts as a Kerberos client when authenticating with Bookie. According to [BookKeeper document](http://bookkeeper.apache.org/docs/latest/security/sasl/), you need to add `bookkeeperClientAuthenticationPlugin` parameter in `broker.conf`:
+Pulsar Broker acts as a Kerberos client when you are authenticating with Bookie. According to [BookKeeper document](http://bookkeeper.apache.org/docs/latest/security/sasl/), you need to add `bookkeeperClientAuthenticationPlugin` parameter in `broker.conf`:
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]: Replace "when you are v-ing" with "when you v xxx".
Use present tense in technical writing. Check all similar cases.

File: site2/docs/security-kerberos.md [hunk 7]
```diff
@@ -346,14 +342,14 @@ superUserRoles=client/{clientIp}@EXAMPLE.COM
...
-Pulsar Broker acts as a Kerberos client when authenticating with Zookeeper. According to [ZooKeeper document](https://cwiki.apache.org/confluence/display/ZOOKEEPER/Client-Server+mutual+authentication), you need these settings in `conf/zookeeper.conf`:
+Pulsar Broker acts as a Kerberos client when you are authenticating with Zookeeper. According to [ZooKeeper document](https://cwiki.apache.org/confluence/display/ZOOKEEPER/Client-Server+mutual+authentication), you need these settings in `conf/zookeeper.conf`:
 
 ```
 authProvider.1=org.apache.zookeeper.server.auth.SASLAuthenticationProvider
 requireClientAuthScheme=sasl
 ```
 
-And add a section of `Client` configurations in the file `pulsar_jaas.conf`, which is used by Pulsar Broker:
+And enter the following commands to add a section of `Client` configurations in the file `pulsar_jaas.conf`, which Pulsar Broker uses:
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]: Delete "And" at the beginning.

File: site2/docs/security-kerberos.md [hunk 8]
```diff
@@ -296,7 +292,7 @@ Pulsar client configuration is similar with client and broker configuration, exc
          .build();
  ```
 
-> The first two lines in the example above are hard coded, alternatively, you can set additional JVM parameters for JAAS and krb5 configuration file when running the application like below:
+> The first two lines in the example above are hard coded, alternatively, you can set additional JVM parameters for JAAS and krb5 configuration file when you are running the application like below:
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]: when you run

File: site2/docs/security-kerberos.md [hunk 9]
```diff
@@ -187,46 +183,46 @@ The following is an example of creating a Java client:
...
-If a client wants to connect to Pulsar Proxy, it is a little different. Client (as a SASL client in Kerberos) will be authenticated by Pulsar Proxy (as a SASL Server in Kerberos) first; and then Pulsar Proxy will be authenticated by Pulsar broker. 
+A client that connects to Pulsar Proxy is a little different. Pulsar Proxy (as a SASL Server in Kerberos) authenticates Client (as a SASL client in Kerberos) first; and then Pulsar broker authenticates Pulsar Proxy. 
 
-Now comparing with the above configuration between client and broker, we will show how to configure Pulsar Proxy. 
+Now in comparision with the above configuration between client and broker, we show you how to configure Pulsar Proxy as follows. 
 
 ### Create principal for Pulsar Proxy in Kerberos
 
-Comparing with the above configuration, you need to add new principal for Pulsar Proxy. If you already have principals for client and broker, only add proxy principal here.
+In comparision with the above configuration, you need to add new principals for Pulsar Proxy. If you already have principals for client and broker, only add the proxy principal here.
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]: re-write this sentence.

File: site2/docs/security-kerberos.md [hunk 10]
```diff
@@ -187,46 +183,46 @@ The following is an example of creating a Java client:
...
 
-Make sure that the keytabs configured in the `pulsar_jaas.conf` file and kdc server in the `krb5.conf` file are reachable by the operating system user who is starting pulsar client.
+You must ensure that the operating system user who is starting pulsar client can reach the keytabs configured in the `pulsar_jaas.conf` file and kdc server in the `krb5.conf` file.
 
 #### Configure CLI tools
 
 If you are using a command-line tool (such as `bin/pulsar-client`, `bin/pulsar-perf` and `bin/pulsar-admin`), you need to preform the following steps:
 
-Step 1. Config your `client.conf`.
+Step 1. Enter the command below to config your `client.conf`.
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]: config-->configure

File: site2/docs/security-kerberos.md [hunk 11]
```diff
@@ -187,46 +183,46 @@ The following is an example of creating a Java client:
...
 
-> The first two lines in the example above are hard coded, alternatively, you can set additional JVM parameters for JAAS and krb5 configuration file when running the application like below:
+> The first two lines in the example above are hard coded, alternatively, you can set additional JVM parameters for JAAS and krb5 configuration file when you are running the application like below:
 
 ```
 java -cp -Djava.security.auth.login.config=/etc/pulsar/pulsar_jaas.conf -Djava.security.krb5.conf=/etc/pulsar/krb5.conf $APP-jar-with-dependencies.jar $CLASSNAME
 ```
 
-Make sure that the keytabs configured in the `pulsar_jaas.conf` file and kdc server in the `krb5.conf` file are reachable by the operating system user who is starting pulsar client.
+You must ensure that the operating system user who is starting pulsar client can reach the keytabs configured in the `pulsar_jaas.conf` file and kdc server in the `krb5.conf` file.
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]: is starting-->starts

File: site2/docs/security-kerberos.md [hunk 12]
```diff
@@ -160,13 +156,13 @@ In client application, include `pulsar-client-auth-sasl` in your project depende
...
 
-There are 2 parameters needed: 
-- `saslJaasClientSectionName` is corresponding to the section in JAAS configuration file for client; 
-- `serverType` stands for whether this client is connect to broker or proxy, and client use this parameter to know which server side principal should be used. 
+You need 2 parameters: 
+- `saslJaasClientSectionName`, which is corresponding to the section in JAAS configuration file for client; 
+- `serverType`, which stands for whether this client is connected to broker or proxy. And client uses this parameter to know this client should use which server side principal. 
 
-When authenticate between client and broker with the setting in above JAAS configuration file, we need to set `saslJaasClientSectionName` to `PulsarClient` and `serverType` to `broker`.
+When you authenticates between client and broker with the setting in above JAAS configuration file, we need to set `saslJaasClientSectionName` to `PulsarClient` and set `serverType` to `broker`.
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]: "you authenticates" or "you authenticate"?

File: site2/docs/security-kerberos.md [hunk 13]
```diff
@@ -160,13 +156,13 @@ In client application, include `pulsar-client-auth-sasl` in your project depende
...
 
-configure the authentication type to use `AuthenticationSasl`, and also provide the authentication parameters to it. 
+Configure the authentication type to use `AuthenticationSasl`, and also provide the authentication parameters to it. 
 
-There are 2 parameters needed: 
-- `saslJaasClientSectionName` is corresponding to the section in JAAS configuration file for client; 
-- `serverType` stands for whether this client is connect to broker or proxy, and client use this parameter to know which server side principal should be used. 
+You need 2 parameters: 
+- `saslJaasClientSectionName`, which is corresponding to the section in JAAS configuration file for client; 
+- `serverType`, which stands for whether this client is connected to broker or proxy. And client uses this parameter to know this client should use which server side principal.
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]: re-write.

File: site2/docs/security-kerberos.md [hunk 14]
```diff
@@ -116,11 +113,11 @@ You can have 2 separate JAAS configuration files:
...
  - Set `saslJaasClientAllowedIds` regex for principal that is allowed to connect to broker;
- - Set `saslJaasBrokerSectionName` that corresponding to the section in JAAS configuration file for broker;
+ - Set `saslJaasBrokerSectionName` that corresponds to the section in JAAS configuration file for broker;
  
  To make Pulsar internal admin client work properly, you need to set the configuration in the `broker.conf` file as below: 
  - Set `brokerClientAuthenticationPlugin` to client plugin `AuthenticationSasl`;
- - Set `brokerClientAuthenticationParameters` to value in JSON string `{"saslJaasClientSectionName":"PulsarClient", "serverType":"broker"}`, in which `PulsarClient` is the section name in above `pulsar_jaas.conf` file, and `"serverType":"broker"` indicate that internal admin client will connect to a Pulsar Broker;
+ - Set `brokerClientAuthenticationParameters` to value in JSON string `{"saslJaasClientSectionName":"PulsarClient", "serverType":"broker"}`, in which `PulsarClient` is the section name in above `pulsar_jaas.conf` file, and `"serverType":"broker"` indicates that the internal admin client connects to a Pulsar Broker;
```
Comment (Jennifer88huang) [inline site2/docs/security-kerberos.md]:
```suggestion
 - Set `brokerClientAuthenticationParameters` to value in JSON string `{"saslJaasClientSectionName":"PulsarClient", "serverType":"broker"}`, in which `PulsarClient` is the section name in the `pulsar_jaas.conf` file, and `"serverType":"broker"` indicates that the internal admin client connects to a Pulsar Broker;
```

**Pull request merged**